### PR TITLE
Add config flags to exclude lines from comments

### PR DESCRIPTION
**Pull request by hedtke**
We use `@exclude` to mark the point from which on the documentation should not be public.

We use the buf linter and do not want to show the linting commands in the public doc.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -78,6 +78,12 @@ footer-content = "© 2023 Jane Doe. All rights reserved."
 repository-type = "github"
 repository-url = "https://github.com/janedoe/myawesomeproject"
 repository-branch = "main"
+
+# In each comment, ignore everything following '@exclude'.
+# Default value: false
+respect-exclude = true
+# In each comment, ignore every line containing 'buf:lint'.
+exclude_buf_lint = true
```
Comment (markvincze) [inline README.md:86]: We should have hyphens here and not underlines, so it should be `exclude-buf-lint`, right?

**Comment by markvincze** [comment]
Hi @hedtke,

Thanks for the submission, looks good! And thanks for the patience.
Could you include the two new fields including their description also in the config example in `README.md`?
And can you also bump the version number in `pyproject.toml` to `0.6`, and add a new entry to `CHANGELOG.md`?

**Comment by hedtke** [comment]
Thank you for the feedback!
I was thinking about a generalization: instead of focusing on a single helper (buf) or single magic exclude string, one could give a list:

`ignore_all_lines_containing`

And

`ignore_everything_after`

**Comment by markvincze** [comment]
@hedtke Yep, that sounds good to me as well.
The only thing is that I'd somehow include the term "comment" in the name of the setting, to express that it only affects processing the comments, and no actual types will be ignored.
Maybe something like `ignore_comment_lines_containing`, and `ignore_comments_after`?

And indeed, sounds good to make these lists of strings, so that multiple values can be specified if needed.

**Comment by hedtke** [comment]
version bumped, changelog updated, options aren't flags but lists now

**Comment by markvincze** [comment]
@hedtke Thanks, I'll merge this one 👍
Please open a separate PR for the package hiding config.

**Pull request merged**